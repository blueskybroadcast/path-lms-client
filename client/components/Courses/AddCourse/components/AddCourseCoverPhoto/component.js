import React from 'react';
import PropTypes from 'prop-types';
import ReactFilestack from 'filestack-react';

const AddCourseCoverPhoto = ({
  handleCoverPhotoUrlToggle, coverPhotoUrl, handleCoverPhotoUrlChange
}) => (
  <fieldset>
    <legend>
      <span>Cover Photo</span>
    </legend>
    <div className="control-group cover-photo-wrapper">
      <label htmlFor="Cover photo">
        Cover photo
        <span className="has-tooltip">
          <span
            className="tooltiped"
            data-toggle="tooltip"
            data-original-title="This photo will be displayed on the homepage or course listing page to entice customers to click and find out more about the course."
          >
            <i className="icon icon-question-circle" />
          </span>
        </span>
      </label>
      <div className="control-pair">
        <div className="controls">
          <label
            className="label-switch"
            tabIndex="0"
            htmlFor="cover_photo"
          >
            <input
              defaultChecked={false}
              onChange={handleCoverPhotoUrlToggle}
              label="false"
              id="cover_photo"
              className="boolean optional switcher optional"
              type="checkbox"
              value="1"
              name="course[cover_photo]"
            />
            <div className="checkbox" />
          </label>
        </div>
        <div className="control-group file optional disabled course_cover_photo_url file-wrapper table-cell">
          <div className="controls">
            { coverPhotoUrl && <span className="upload-completed">Upload Complete</span> }
            { coverPhotoUrl === '' && (
              <ReactFilestack
                apikey={process.env.FILEPICKER_API_KEY}
                buttonText="Choose File"
                buttonClass="btn btn-link"
                disabled
                options={{
                  accept: 'image/*',
                  maxFiles: 1
                }}
                onSuccess={handleCoverPhotoUrlChange}
              />
            )}
          </div>
        </div>
      </div>
      <div className="controls hint-container">
        <p className="hint">Optimal photo size: 320px wide and 152px tall</p>
      </div>
    </div>
  </fieldset>
);

AddCourseCoverPhoto.propTypes = {
  handleCoverPhotoUrlToggle: PropTypes.func.isRequired,
  coverPhotoUrl: PropTypes.string,
  handleCoverPhotoUrlChange: PropTypes.func.isRequired
};

AddCourseCoverPhoto.defaultProps = {
  coverPhotoUrl: undefined
};

export default AddCourseCoverPhoto;
