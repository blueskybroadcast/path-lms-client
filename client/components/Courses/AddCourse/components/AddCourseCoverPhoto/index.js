import AddCourseCoverPhoto from './component';

export default AddCourseCoverPhoto;
