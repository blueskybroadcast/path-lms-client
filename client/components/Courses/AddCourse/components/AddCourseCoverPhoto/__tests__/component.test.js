import React from 'react';
import { shallow } from 'enzyme';
import AddCourseCoverPhoto from '../component';

describe('AddCourseCoverPhoto', () => {
  test('should render AddCourseCoverPhoto with cover photo url correctly', () => {
    const wrapper = shallow(
      <AddCourseCoverPhoto
        coverPhotoUrl="cover_photo_url"
        handleCoverPhotoUrlToggle={jest.fn()}
        handleCoverPhotoUrlChange={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render AddCourseCoverPhoto without cover photo correctly', () => {
    const wrapper = shallow(
      <AddCourseCoverPhoto
        handleCoverPhotoUrlToggle={jest.fn()}
        handleCoverPhotoUrlChange={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
