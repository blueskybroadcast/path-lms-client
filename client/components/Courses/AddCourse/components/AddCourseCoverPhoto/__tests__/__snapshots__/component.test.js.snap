// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseCoverPhoto should render AddCourseCoverPhoto with cover photo url correctly 1`] = `
<fieldset>
  <legend>
    <span>
      Cover Photo
    </span>
  </legend>
  <div
    className="control-group cover-photo-wrapper"
  >
    <label
      htmlFor="Cover photo"
    >
      Cover photo
      <span
        className="has-tooltip"
      >
        <span
          className="tooltiped"
          data-original-title="This photo will be displayed on the homepage or course listing page to entice customers to click and find out more about the course."
          data-toggle="tooltip"
        >
          <i
            className="icon icon-question-circle"
          />
        </span>
      </span>
    </label>
    <div
      className="control-pair"
    >
      <div
        className="controls"
      >
        <label
          className="label-switch"
          htmlFor="cover_photo"
          tabIndex="0"
        >
          <input
            className="boolean optional switcher optional"
            defaultChecked={false}
            id="cover_photo"
            label="false"
            name="course[cover_photo]"
            onChange={[MockFunction]}
            type="checkbox"
            value="1"
          />
          <div
            className="checkbox"
          />
        </label>
      </div>
      <div
        className="control-group file optional disabled course_cover_photo_url file-wrapper table-cell"
      >
        <div
          className="controls"
        >
          <span
            className="upload-completed"
          >
            Upload Complete
          </span>
        </div>
      </div>
    </div>
    <div
      className="controls hint-container"
    >
      <p
        className="hint"
      >
        Optimal photo size: 320px wide and 152px tall
      </p>
    </div>
  </div>
</fieldset>
`;

exports[`AddCourseCoverPhoto should render AddCourseCoverPhoto without cover photo correctly 1`] = `
<fieldset>
  <legend>
    <span>
      Cover Photo
    </span>
  </legend>
  <div
    className="control-group cover-photo-wrapper"
  >
    <label
      htmlFor="Cover photo"
    >
      Cover photo
      <span
        className="has-tooltip"
      >
        <span
          className="tooltiped"
          data-original-title="This photo will be displayed on the homepage or course listing page to entice customers to click and find out more about the course."
          data-toggle="tooltip"
        >
          <i
            className="icon icon-question-circle"
          />
        </span>
      </span>
    </label>
    <div
      className="control-pair"
    >
      <div
        className="controls"
      >
        <label
          className="label-switch"
          htmlFor="cover_photo"
          tabIndex="0"
        >
          <input
            className="boolean optional switcher optional"
            defaultChecked={false}
            id="cover_photo"
            label="false"
            name="course[cover_photo]"
            onChange={[MockFunction]}
            type="checkbox"
            value="1"
          />
          <div
            className="checkbox"
          />
        </label>
      </div>
      <div
        className="control-group file optional disabled course_cover_photo_url file-wrapper table-cell"
      >
        <div
          className="controls"
        />
      </div>
    </div>
    <div
      className="controls hint-container"
    >
      <p
        className="hint"
      >
        Optimal photo size: 320px wide and 152px tall
      </p>
    </div>
  </div>
</fieldset>
`;
